server.go

```diff
@@ -133,52 +133,121 @@ func (app *Yarana) replyText(replyToken, text string) error {
+		if err != nil {
+			return err
+		}
+	case RequstTypeGetActivities:
+		err = app.processGetActivities(replyToken, source.UserID, userReq.VariableKeyword)
+		if err != nil {
+			return err
+		}
+	case RequstTypeAddActivity:
+		err = app.processAddActivity(replyToken, source.UserID, userReq.VariableKeyword)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
 
+func (app *Yarana) handleImage(message *linebot.ImageMessage, replyToken string) error {
+	return nil
+}
+
+func (app *Yarana) processGetKotos(replyToken string, userID string, keyword string) error {
 	// Get Kotos
-	kotos, err := app.dataCall.GetKotosByUserID("d59964bb713fd6f4f5ef6a7c7e029387") // sample data in Cosmos DB
+	kotos, err := app.dataCall.GetKotosByUserID(userID)
 	if err != nil {
 		return err
 	}
 
-	// Add Koto
-	kotoToAdd, _ := NewKotoData("", source.UserID, "Demo AddKoto")
-	err = app.dataCall.AddKoto(kotoToAdd)
+	// Make text to send
+	var textToSend string
+	if len(kotos) == 0 || kotos == nil {
+		textToSend = "No Koto Data"
+	} else {
+		textToSend = kotos[0].Title
+	}
+	if _, err := app.bot.ReplyMessage(
+		replyToken,
+		linebot.NewTextMessage(textToSend),
+	).Do(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (app *Yarana) processAddKoto(replyToken string, userID string, keyword string) error {
+	kotoToAdd, _ := NewKotoData("", userID, keyword)
+	errChan := make(chan error, 1)
+
+	// Add Koto Data
+	go func() {
+		err := app.dataCall.AddKoto(kotoToAdd)
+		errChan <- err
+	}()
+
+	var textToSend string
+
+	err := <-errChan
 	if err != nil {
+		textToSend = "I'm sorry I failed to add your new やること."
+		if _, err := app.bot.ReplyMessage(
+			replyToken,
+			linebot.NewTextMessage(textToSend),
+		).Do(); err != nil {
+			return err
+		}
+		return err // TODO: I wonder the `err` scope. app.bot.ReplyMessage error or app.dataCall.AddKoto error?
+	}
+	textToSend = "I added your new やること"
+	if _, err := app.bot.ReplyMessage(
+		replyToken,
+		linebot.NewTextMessage(textToSend),
+	).Do(); err != nil {
 		return err
 	}
 
-	// Get Activities
+	return nil
+}
+
+func (app *Yarana) processGetActivities(replyToken string, userID string, keyword string) error {
 	/*
 		acts, err := app.dataCall.GetActivitiesByKotoDataID("123456789") // TODO: test code
 		if err != nil {
 			return err
 		}
 		actsString := fmt.Sprint(acts)
 	*/
+	return nil
+}
 
-	// Make text to send
-	var textToSend string
-	// textToSend = message.Text // That's "Oumugaeshi"
-	// textToSend := actsString
-	if len(kotos) == 0 || kotos == nil {
-		textToSend = "No Koto Data"
-	} else {
-		textToSend = kotos[0].Title
-	}
+func (app *Yarana) processAddActivity(replyToken string, userID string, keyword string) error {
+	return nil
+}
 
+func (app *Yarana) sorryAndShowHelp(replyToken string) error {
 	if _, err := app.bot.ReplyMessage(
 		replyToken,
-		linebot.NewTextMessage(textToSend),
+		linebot.NewTextMessage("I'm sorry that's invalid input for me."), // TODO: Show HELP View to user
```

Comment:
We have to consider behavior of reply message when the bot got incomprehensible message.